### ISSUE TITLE
RSS Feeds on apps client

### ISSUE THREAD

**Issue by jac110** [igniterealtime/Pade#192]
Hello,

New to this community and happy to contribute ;)

I'm planning a rollout of the web version of Pàdé (apps). The RSS Feeds looks like a promising  tool for my setup, but I fail to make it work. 

Just to make sure that I do things properly :

- find the RSS feed or url (view page source). For instance:
https://discourse.igniterealtime.org/c/blogs.rss
or in my case:
https://unblogavecvue.com/feed
- tick the Feeds boxes in the settings of Pàdé;
Enable RSS and Atom Feeds
Show feed summary
- copy the feed urls in the form
- define the interval refresh time
- F5 seetings and Pàdé

I fail to see the headlines under RSS Feeds. 
I'm using Pàdé apps ver 1.2.8 via Edge dev as a pwa app if this matters.

Thanks in advance, 
Jacques

**Comment by deleolajide** [comment]
> New to this community and happy to contribute ;)

Thank you. Any contribution is most welcome :-)

>I fail to see the headlines under RSS Feeds. I'm using Pàdé apps ver 1.2.8 via Edge dev as a pwa app if this matters.

I am afraid the RSS Feeds won't work from a standard web page without support of a proxy web server because cross-domain permissions and policies. In other words, the JavaScript in the web page from your web site cannot load an RSS/Atom file from a web site with a different domain.

Pade as a browser extension does not have this limitation because the browser extension JavaScript is trusted.

**Comment by deleolajide** [comment]
The best way of using the RSS Feed is to add it to a groupchat. For example, I use it with the conversejs.org github commit feed by issuing a /feed command like this in the converse groupchat room
```
/feed https://github.com/conversejs/converse.js/commits/master.atom
```

This is just like the /feed command in slack

**Comment by jac110** [comment]
Thank you Dele for your prompt answer.

I shall test it and whenever possible reconsider the browser extension as an alternative for my rollout.

**Comment by jac110** [comment]
> In other words, the JavaScript in the web page from your web site cannot load an RSS/Atom file from a web site with a different domain.

As a workaround, what about redirecting Pàdé (the current xmpp server) to one of my (sub) domains, i.e. the same domain in which I plan to produce RSS compatible post and content ?

**Comment by deleolajide** [comment]
The main index.html page that loads the Pàdé Javascript code has to be fetched from web server that fetches pages for the same domain in which you plan to fetch the RSS content. That will work for that specific case as a work-around, but you have to do the same thing for every other domain that serves RSS/Atom content.

I am close to providing a proxy RSS/Atom fetch service directly from Openfire that will act just like the browser extension privilege. See https://github.com/igniterealtime/Openfire-Chat/issues/13

**Comment by deleolajide** [comment]
This is now available in Openfire Chat API plugin version 0.9.5 Release 5

![image](https://user-images.githubusercontent.com/110731/60434058-29fdad00-9bfe-11e9-9278-82c9a940a9f6.png)
